### PR TITLE
Add spatialite uri support in bad layer handler

### LINKED ISSUE #41690: Handle Unavalable Layers acting unexpectedly
On 

QGIS version | 3.16.3-Hannover | QGIS code branch | Release 3.16
-- | -- | -- | --
Compiled against Qt | 5.15.2 | Running against Qt | 5.15.2
Compiled against GDAL/OGR | 3.0.4 | Running against GDAL/OGR | 3.0.4
Compiled against GEOS | 3.8.1-CAPI-1.13.3 | Running against GEOS | 3.8.1-CAPI-1.13.3
Compiled against SQLite | 3.34.1 | Running against SQLite | 3.34.1
PostgreSQL Client Version | 13.1 | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.6 | QScintilla2 Version | 2.11.6
Compiled against PROJ | 6.3.2 | Running against PROJ | Rel. 6.3.2, May 1st, 2020
OS Version | Manjaro Linux
Active python plugins | TomBio; db_manager; processing

When opening a project where the main data-source (a spatialite database) has changed name, the 'Handle Unavailable Layers' process is acting unexpectedly. Firstly it's not registering the new name when I correct it, it continues to treat it as an unavailable source even though it's now a bona fide source in the correct directory. Secondly, for three shapefiles that are also part of the project (but needed renaming too), it has simply automatically selected one of the shapefiles in the directory to match all three. Thirdly, when finally selecting 'keep unavailable layers' each unavailable layer is assigned, as a data-source, the entire unavailable source urls all concatenated together into one long text string.

When I repair data-source in the project, it works for one or two layers without fault and then QGIS crashes.

### PR DESCRIPTION
**Pull request by roya0045**
## Description

 Fixes #41690

Need to do some string fiddling to get the path and give it back just right to please the provider...

Tester for apply on manually fixed name and autofind

### CODE REVIEW

File: src/app/qgshandlebadlayers.cpp [hunk 1]
```diff
@@ -383,12 +386,23 @@ void QgsHandleBadLayers::apply()
     QTableWidgetItem *item = mLayerList->item( i, 4 );
     QString datasource = item->text();
     QString fileName;
+    QString formatter;
     const QString layerId { node.namedItem( QStringLiteral( "id" ) ).toElement().text() };
     const QString name { mLayerList->item( i, 0 )->text() };
+    QString provider = node.namedItem( QStringLiteral( "provider" ) ).toElement().text();
+    if ( provider == QString( "spatialite" ) )
+    {
+        if ( datasource.contains( "dbname='" ) )
```
Comment (nyalldawson) [inline src/app/qgshandlebadlayers.cpp:395]: Could the provider decodeUri api be used here instead of the raw string handling?

Comment (roya0045) [reply src/app/qgshandlebadlayers.cpp:395]: In my test I had bad results with it but give me a moment I'll try something else.

Comment (roya0045) [reply src/app/qgshandlebadlayers.cpp]: reworked the uri handling to use the raw string instead of an altered string, other behavior seems unchanged.

**Comment by roya0045** [comment]
Backport seems adequate.

**Comment by roya0045** [comment]
Superseded by #42154

**Pull request closed without merging**